### PR TITLE
Ace3, settings, and various visual improvements

### PR DESCRIPTION
**Pull request by xvwyh**
WIP so far:
image

TODO:
- Settings button providing options to:
  - Lock the frame's position and size
  - Make the backdrop always visible instead of just on hover
  - Hide the frame entirely when no VO is being played

### CODE REVIEW

**Comment by cralor** [comment]
Great additions! I think it would be great to add a border around the talking head. Could even remove the "list" of queue and make it a 1/X queue like Immersion does, with the X to skip the current sound.

image

**Comment by xvwyh** [comment]
I'll let MrThinger decide where he wants to take the UI going forward. My job here is just to make a basic usable frame with minor customization that isn't a big invisible rectangle that blocks mouse clicks.

**Comment by xvwyh** [comment]
Threw together a small dropdown menu for frame settings just to allow basic customization until something better is created.
image

Now, `VoiceOverSettings` is just a temporary solution to store these settings, but I would reeeeally not recommend keeping it. I would strongly encourage switching over to Ace3 framework and using its AceDB/AceConfig for settings and AceAddon/AceEvent for event handling instead of VoiceOver's own EventHandler. I can sort it all out if I get a green light.

File: VoiceOver/SoundQueueUI.lua [hunk 1]
```diff
@@ -16,49 +16,131 @@ function SoundQueueUI:new(soundQueue)
...
     self.soundQueueFrame:SetWidth(300)
     self.soundQueueFrame:SetHeight(300)
     self.soundQueueFrame:SetPoint("BOTTOMRIGHT", 0, 0)
     self.soundQueueFrame.buttons = {}
     self.soundQueueFrame:SetMovable(true)  -- Allow the frame to be moved
-    self.soundQueueFrame:EnableMouse(true) -- Allow the frame to be clicked on
+    self.soundQueueFrame:SetResizable(true)  -- Allow the frame to be resized
+    self.soundQueueFrame:SetClampedToScreen(true) -- Prevent from being dragged off-screen
+    self.soundQueueFrame:SetUserPlaced(true)
+    self.soundQueueFrame:SetResizeBounds(200, 40 + 64 + 10)
```
Comment (cralor) [inline VoiceOver/SoundQueueUI.lua:36]: 1x Interface\AddOns\VoiceOver\SoundQueueUI.lua:36: attempt to call method 'SetResizeBounds' (a nil value)
[string "@Interface\AddOns\VoiceOver\SoundQueueUI.lua"]:36: in function `initDisplay'
[string "@Interface\AddOns\VoiceOver\SoundQueueUI.lua"]:15: in function `new'
[string "@Interface\AddOns\VoiceOver\SoundQueue.lua"]:9: in function `new'
[string "@Interface\AddOns\VoiceOver\Main.lua"]:5: in main chunk

Locals:
self = <table> {
 soundQueueScrollFrame = <unnamed> {
 }
 animtimer = 1680021736
 soundQueue = <table> {
 }
 soundQueueButtonContainer = <unnamed> {
 }
 soundQueueFrame = VoiceOverSoundQueueFrame {
 }
}
(*temporary) = nil
(*temporary) = VoiceOverSoundQueueFrame {
 0 = <userdata>
 OnBackdropLoaded = <function> defined @Interface\SharedXML\Backdrop.lua:144
 GetBackdropColor = <function> defined @Interface\SharedXML\Backdrop.lua:382
 SetupTextureCoordinates = <function> defined @Interface\SharedXML\Backdrop.lua:206
 OnBackdropSizeChanged = <function> defined @Interface\SharedXML\Backdrop.lua:174
 HasBackdropInfo = <function> defined @Interface\SharedXML\Backdrop.lua:270
 SetBackdropBorderColor = <function> defined @Interface\SharedXML\Backdrop.lua:414
 GetEdgeSize = <function> defined @Interface\SharedXML\Backdrop.lua:180
 buttons = <table> {
 }
 ApplyBackdrop = <function> defined @Interface\SharedXML\Backdrop.lua:286
 SetBackdrop = <function> defined @Interface\SharedXML\Backdrop.lua:321
 ClearBackdrop = <function> defined @Interface\SharedXML\Backdrop.lua:274
 GetBackdropBorderColor = <function> defined @Interface\SharedXML\Backdrop.lua:401
 GetBackdrop = <function> defined @Interface\SharedXML\Backdrop.lua:339
 SetupPieceVisuals = <function> defined @Interface\SharedXML\Backdrop.lua:238
 SetBackdropColor = <function> defined @Interface\SharedXML\Backdrop.lua:391
 SetBorderBlendMode = <function> defined @Interface\SharedXML\Backdrop.lua:258
}
(*temporary) = 200
(*temporary) = 114
(*temporary) = "attempt to call method 'SetResizeBounds' (a nil value)"

Comment (xvwyh) [reply VoiceOver/SoundQueueUI.lua:36]: 1.14?

Comment (cralor) [reply VoiceOver/SoundQueueUI.lua:36]: Yep

**Comment by mrthinger** [comment]
Fine transitioning to ace

File: VoiceOver/SoundQueueUI.lua (hunk 1 above)
Comment (xvwyh) [reply VoiceOver/SoundQueueUI.lua]: Should hopefully be fixed now

**Pull request merged**